src/encoded/static/components/rnaget.js

```diff
@@ -0,0 +1,889 @@
+    }
+
+    handleInputChange(e) {
+        this.setState({ genes: e.target.value });
+    }
+
+    handleSubmit(e) {
+        e.preventDefault();
+        window.location.href = `/rnaget?genes=${this.state.genes}&units=${this.state.unitsOption}`;
+    }
+
+    render() {
+        return (
+            <form onSubmit={this.handleSubmit}>
+                <div className="rna_seq_matrix-search">
+                    <input
+                        type="search"
+                        className="search-query"
+                        placeholder="Enter Gene IDs..."
+                        value={this.state.genes}
+                        onKeyDown={this.onKeyDown}
+                        onChange={this.handleInputChange}
+                    />
+                    <select name="searchOption" onChange={this.handleChange} value={this.state.unitsOption}>
+                        <option value="tpm">TPM</option>
+                        <option value="fpkm">FPKM</option>
+                    </select>
+                    <button type="submit">Search</button>
+                </div>
+            </form>
+        );
+    }
+}
+
+
+/**
+ * Displays a control allowing the user to select the maximum number of items to display on one page.
+ */
+const pageLimitOptions = [25, 50, 100, 200];
+const PageLimitSelector = ({ pageLimit, query }) => (
+    <div className="page-limit-selector">
+        <div className="page-limit-selector__label">Items per page:</div>
+        <div className="page-limit-selector__options">
+            {pageLimitOptions.map((limit) => {
+                // When changing the number of items per page, also go back to the first page by
+                // removing the "from=x" query-string parameter.
+                const limitQuery = query.clone();
+                limitQuery.deleteKeyValue('from');
+                if (limit === DEFAULT_PAGE_LIMIT) {
+                    limitQuery.deleteKeyValue('limit');
+                } else {
+                    limitQuery.replaceKeyValue('limit', limit);
+                }
+
+                return (
+                    <a
+                        key={limit}
+                        href={`?${limitQuery.format()}`}
+                        className={`page-limit-selector__option${limit === pageLimit ? ' page-limit-selector__option--selected' : ''}`}
+                        aria-label={`Show ${limit} items per page`}
+                    >
+                        {limit}
+                    </a>
+                );
+            })}
+        </div>
+    </div>
+);
+
+PageLimitSelector.propTypes = {
+    /** New page limit to display */
+    pageLimit: PropTypes.number.isRequired,
+    /** Current page's QueryString query */
+    query: PropTypes.object.isRequired,
+};
+
+/**
+ * Renderer for the entire /report/ page.
+ */
+const RNAGet = ({ context }, reactContext) => {
+    /** All possible columns for the current type */
+    const [allColumns] = React.useState(null);
```

Comment:
Does this ever need updating? It's rhetorical as the code tells me know albeit it seems odd.